### PR TITLE
Cleaning up cmake

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: python/source/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,11 @@ SET_TARGET_PROPERTIES(pycap PROPERTIES
     SUFFIX ".so"
     )
 
-TARGET_LINK_LIBRARIES(pycap ${Boost_LIBRARIES} ${PYTHON_LIBRARIES} cap)
+TARGET_LINK_LIBRARIES(pycap cap)
+TARGET_LINK_LIBRARIES(pycap ${Boost_MPI_LIBRARY})
+TARGET_LINK_LIBRARIES(pycap ${Boost_SERIALIZATION_LIBRARY})
+TARGET_LINK_LIBRARIES(pycap ${Boost_PYTHON_LIBRARY})
+TARGET_LINK_LIBRARIES(pycap ${PYTHON_LIBRARIES})
```
Comment (Rombur) [inline python/source/CMakeLists.txt:28]: Are you sure you didn't forget some libraries? I had this error before:
`ImportError: libboost_timer.so.1.60.0: cannot open shared object file: No such file or directory`
 I fixed it by exporting the whole boost library

**Comment by dalg24** [comment]
not sure no

**Comment by Rombur** [comment]
here is the list that _pycap requires on my install:
libboost_iostreams.so.1.60.0
libboost_system.so.1.60.0
libboost_thread.so.1.60.0
libboost_python.so.1.60.0 
libboost_unit_test_framework.so.1.60.0 
libboost_serialization.so.1.60.0 
libboost_mpi.so.1.60.0 
It also requires all the trilinos packages that I have installed, deal.II, and p4est.
If you create a shared library for libcap it will be easier to find out what it needs.

**Comment by Rombur** [comment]
Error when compiling:

/home/bt2/software/boost_1_60_0/install/lib/libboost_python.so: undefined reference to `PySlice_New'
/home/bt2/software/boost_1_60_0/install/lib/libboost_python.so: undefined reference to`PyList_Sort'
/home/bt2/software/boost_1_60_0/install/lib/libboost_python.so: undefined reference to `PySequence_GetSlice'
/home/bt2/software/boost_1_60_0/install/lib/libboost_python.so: undefined reference to`PyType_IsSubtype'
/home/bt2/software/boost_1_60_0/install/lib/libboost_python.so: undefined reference to `PyExc_RuntimeError'

**Comment by dalg24** [comment]
libcap compiled just fine?
did that occur when compiling _pycap.so?

**Comment by Rombur** [comment]
Yes it looks like libcap compiled.

**Comment by Rombur** [comment]
error while loading shared libraries: libboost_chrono.so.1.60.0

**Comment by dalg24** [comment]
awww man

**Comment by Rombur** [comment]
it should be enough to add chrono. chrono only depends on system.

**Comment by Rombur** [comment]
It works :-)

**Pull request merged**